### PR TITLE
Feature/task feedback

### PR DESCRIPTION
**Pull request by james-toussaint**

### CODE REVIEW

File: src/main/java/com/iexec/common/replicate/ReplicateStatusCause.java [hunk 1]
```diff
@@ -40,7 +40,21 @@ public enum ReplicateStatusCause {
...
+    PRE_COMPUTE_INVALID_DATASET_CHECKSUM,
+    PRE_COMPUTE_INVALID_DATASET_KEY,
+    PRE_COMPUTE_DATASET_DECRYPTION_FAILED,
+    PRE_COMPUTE_SAVING_PLAIN_DATASET_FAILED,
+    PRE_COMPUTE_BAD_INPUT_FILE_ARGS,
+    PRE_COMPUTE_INPUT_FILE_DOWNLOAD_FAILED, //exit 1 known end
+    PRE_COMPUTE_UNKNOWN_FAILED, //exit 1 unknown
+    PRE_COMPUTE_NO_REPORT_FAILED, //exit 2
+    PRE_COMPUTE_NO_CONTEXT_FAILED, //exit 3
+    // post-computation
```
Comment (mcornaton) [inline src/main/java/com/iexec/common/replicate/ReplicateStatusCause.java]: Are you sure all of the following values are linked to Post-Compute?

File: src/main/java/com/iexec/common/replicate/ReplicateStatusCause.java [hunk 2]
```diff
@@ -40,15 +40,52 @@ public enum ReplicateStatusCause {
...
+    POST_COMPUTE_MISSING_STORAGE_TOKEN,
+    POST_COMPUTE_COMPUTED_FILE_NOT_FOUND,
+    POST_COMPUTE_RESULT_DIGEST_COMPUTATION_FAILED,
+    POST_COMPUTE_OUT_FOLDER_ZIP_FAILED,
+    POST_COMPUTE_ENCRYPTION_FAILED,
+    POST_COMPUTE_RESULT_FILE_NOT_FOUND,
+    POST_COMPUTE_DROPBOX_UPLOAD_FAILED,
+    POST_COMPUTE_IPFS_UPLOAD_RAILED,
+    POST_COMPUTE_INVALID_TEE_SIGNATURE,
+    POST_COMPUTE_SEND_COMPUTED_FILE_FAILED,
```
Comment (jbern0rd) [inline src/main/java/com/iexec/common/replicate/ReplicateStatusCause.java:78]: Several feedbacks on the Enum.
It lacks a bit of consistency between POST and PRE compute.
Some examples:
`PRE_COMPUTE_FAILED_UNKNOWN_ISSUE`, `POST_COMPUTE_UNKNOWN_ISSUE`
`POST_COMPUTE_MISSING_TASK_ID`, `PRE_COMPUTE_TASK_ID_MISSING`
`POST_COMPUTE_OUT_FOLDER`, `PRE_COMPUTE_OUTPUT_FOLDER`
...

I'm ok with either option as long as we have a better consistency and a kind of naming convention.

Small typo in `POST_COMPUTE_IPFS_UPLOAD_RAILED`

File: src/main/java/com/iexec/common/replicate/ReplicateStatusDetails.java [hunk 3]
```diff
@@ -36,6 +36,7 @@ public class ReplicateStatusDetails {
     private String errorMessage;
     private ReplicateStatusCause cause;
     private String stdout;
+    private Integer exitCode; //null means unset
```
Comment (jbern0rd) [inline src/main/java/com/iexec/common/replicate/ReplicateStatusDetails.java:39]:
```suggestion
    private int exitCode; //null means unset
```

Comment (james-toussaint) [reply src/main/java/com/iexec/common/replicate/ReplicateStatusDetails.java:39]: Null `exitCode` means container wasn't even launched, thus `Integer`

File: src/main/java/com/iexec/common/replicate/ReplicateStatusCause.java (hunk 2 above)
Comment (james-toussaint) [reply src/main/java/com/iexec/common/replicate/ReplicateStatusCause.java:78]: Yes, I agree and plan to update, thanks for reminding it :)

**Review by jbern0rd — APPROVED** [review]

**Review by mcornaton — APPROVED** [review]

File: src/main/java/com/iexec/common/replicate/ReplicateStatusDetails.java (hunk 3 above)
Comment (james-toussaint) [reply src/main/java/com/iexec/common/replicate/ReplicateStatusDetails.java:39]: May I close it @jbern0rd?

File: src/main/java/com/iexec/common/replicate/ReplicateStatusCause.java (hunk 2 above)
Comment (james-toussaint) [reply src/main/java/com/iexec/common/replicate/ReplicateStatusCause.java:78]: Thanks for pointing them out, @mcornaton updated them. Closing this comment.

File: src/main/java/com/iexec/common/replicate/ReplicateStatusDetails.java (hunk 3 above)
Comment (jbern0rd) [reply src/main/java/com/iexec/common/replicate/ReplicateStatusDetails.java:39]: May you close what ?
I have approved since last week

Comment (james-toussaint) [reply src/main/java/com/iexec/common/replicate/ReplicateStatusDetails.java:39]: You globally set approval but did not add any comment/thumb-up nor resolved the conversation you raised, that is why I preferred to ask.
Sorry for your time if I bothered you, closing now.

Comment (jbern0rd) [reply src/main/java/com/iexec/common/replicate/ReplicateStatusDetails.java:39]: sorry, I should have resolved this thread

File: src/main/java/com/iexec/common/worker/api/ExitMessage.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2020 IEXEC BLOCKCHAIN TECH
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/worker/api/ExitMessage.java]:
```suggestion
 * Copyright 2022 IEXEC BLOCKCHAIN TECH
```

**Pull request merged**